main/internal/implementations/terraform_workspace/azure_backend.go

```diff
@@ -0,0 +1,113 @@
+
+	// workspaceToBackendDetails is a map of Terraform workspace names to their respective backend details.
+	workspaceToBackendDetails map[string]interface{}
+}
+
+// FindTerraformWorkspaces returns a map of Terraform workspace names to their respective directories.
+func (b *AzureBlobBackend) FindTerraformWorkspaces(ctx context.Context) (map[string]string, error) {
+	workspaces, workspaceToBackendDetails, err := findTerraformWorkspaces(ctx, b.dragonDrop, b.config.WorkspaceDirectories, "azure_blob")
+	if err != nil {
+		return nil, err
+	}
+	b.workspaceToBackendDetails = workspaceToBackendDetails
+
+	return workspaces, err
+}
+
+// NewAzureBlobBackend creates a new AzureBlobBackend instance.
+func NewAzureBlobBackend(ctx context.Context, config ContainerBackendConfig, dragonDrop interfaces.DragonDrop) interfaces.TerraformWorkspace {
+	dragonDrop.PostLog(ctx, "Created TFWorkspace client.")
+
+	return &AzureBlobBackend{config: config, dragonDrop: dragonDrop}
+}
+
+// DownloadWorkspaceState downloads from the remote Azure Blob Storage backend the latest state file
+func (b *AzureBlobBackend) DownloadWorkspaceState(ctx context.Context, workspaceToDirectory map[string]string) error {
+	b.dragonDrop.PostLog(ctx, "Beginning download of state files to local memory.")
+
+	for workspaceName := range workspaceToDirectory {
+		err := b.getWorkspaceStateByTestingAllAzureCredentials(ctx, workspaceName)
+		if err == nil {
+			break
+		}
+	}
+
+	b.dragonDrop.PostLog(ctx, "Done with download of state files to local memory.")
+	return nil
+}
+
+// getWorkspaceStateByTestingAllAzureCredentials downloads the state file for the given workspace from the Azure Blob Storage backend.
+func (b *AzureBlobBackend) getWorkspaceStateByTestingAllAzureCredentials(ctx context.Context, workspaceName string) error {
+	for _, credential := range b.config.DivisionCloudCredentials {
+		serviceURL, err := b.configureAzureBlobURL(ctx, credential, b.workspaceToBackendDetails[workspaceName].(BackendAzure))
+		if err != nil {
+			continue
+		}
+
+		stateFileName := fmt.Sprintf("%v.json", workspaceName)
+
+		fileOutPath := fmt.Sprintf("state_files/%v", stateFileName)
+
+		outFile, err := os.Create(fileOutPath)
+		if err != nil {
+			continue
+		}
+
+		blobURL := serviceURL.NewContainerURL(b.config.ContainerName).NewBlobURL(stateFileName)
+
+		err = azblob.DownloadBlobToFile(ctx, blobURL, 0, azblob.CountToEnd, outFile, azblob.DownloadFromBlobOptions{})
+		if err != nil {
+			outFile.Close()
+			continue
+		}
+
+		outFile.Close()
+	}
+
+	return nil
+}
+
+// AzureCredentials is a struct that holds the credentials for an Azure Blob Storage backend.
+type AzureCredentials struct {
+	AzureStorageAccountKey string `json:"azure_storage_account_key"`
+}
+
+// configureAzureBlobURL configures the Azure Blob Storage URL.
+func (b *AzureBlobBackend) configureAzureBlobURL(ctx context.Context, credential terraformValueObjects.Credential, backendAzure BackendAzure) (azblob.ServiceURL, error) {
```

Comment:
Let's add a unit test for this method to ensure that the url format being generated is as expected